src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -14,43 +14,142 @@ internal static class CompletionHelpers
+                    yield return new CompletionResult(completionText, listItemText, resultType, toolTip);
                 }
             }
         }
 
+        /// <summary>
+        /// Normalizes the input string to an expandable string format for PowerShell.
+        /// </summary>
+        /// <param name="value">The input string to be normalized.</param>
+        /// <returns>The normalized string with special characters replaced by their PowerShell escape sequences.</returns>
+        /// <remarks>
+        /// This method replaces special characters in the input string with their PowerShell equivalent escape sequences:
+        /// <list type="bullet">
+        ///     <item><description>Replaces "\r" (carriage return) with "`r".</description></item>
+        ///     <item><description>Replaces "\n" (newline) with "`n".</description></item>
+        ///     <item><description>Replaces "\t" (tab) with "`t".</description></item>
+        ///     <item><description>Replaces "\0" (null) with "`0".</description></item>
+        ///     <item><description>Replaces "\a" (bell) with "`a".</description></item>
+        ///     <item><description>Replaces "\b" (backspace) with "`b".</description></item>
+        ///     <item><description>Replaces "\u001b" (escape character) with "`e".</description></item>
+        ///     <item><description>Replaces "\f" (form feed) with "`f".</description></item>
+        ///     <item><description>Replaces "\v" (vertical tab) with "`v".</description></item>
+        /// </list>
+        /// </remarks>
+        internal static string NormalizeToExpandableString(string value)
+            => value
+                .Replace("\r", "`r")
+                .Replace("\n", "`n")
+                .Replace("\t", "`t")
+                .Replace("\0", "`0")
+                .Replace("\a", "`a")
+                .Replace("\b", "`b")
+                .Replace("\u001b", "`e")
+                .Replace("\f", "`f")
+                .Replace("\v", "`v");
+
+        /// <summary>
+        /// Defines a strategy for determining if a value matches a word or pattern.
+        /// </summary>
+        /// <param name="value">The input string to check for a match.</param>
+        /// <param name="wordToComplete">The word or pattern to match against.</param>
+        /// <returns>
+        /// <c>true</c> if the value matches the specified word or pattern; otherwise, <c>false</c>.
+        /// </returns>
+        internal delegate bool MatchStrategy(string value, string wordToComplete);
+
+        /// <summary>
+        /// Determines if the given value matches the specified word using a literal, case-insensitive prefix match.
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if the value starts with the word (case-insensitively); otherwise, <c>false</c>.
+        /// </returns>
+        internal static readonly MatchStrategy LiteralMatch = (value, wordToComplete)
```

Comment:
If we want to share this maybe rename to reflect OrdinalIgnoreCase.

Reply:
Fixed in 7125ec9f13d656af63fa2d9be897fe51650e5dde